File: coding_bootcamp_v1/examples/languages/go/main.go
package main

import (
    "fmt"
    "strings"
)

func main() {
    text := "go word frequency starter"
    counts := map[string]int{}
    for _, word := range strings.Fields(text) {
        counts[word]++
    }
    for word, count := range counts {
        fmt.Printf("%s: %d\n", word, count)
    }
}

File: coding_bootcamp_v1/examples/languages/kotlin/Main.kt
fun main() {
    val text = "kotlin word frequency starter"
    val counts = mutableMapOf<String, Int>()
    for (word in text.split(Regex("\\s+"))) {
        counts[word] = (counts[word] ?: 0) + 1
    }
    for ((word, count) in counts) {
        println("$word: $count")
    }
}
